lms/djangoapps/certificates/signals.py

```diff
@@ -1,14 +1,35 @@
 """
 Signal handler for enabling/disabling self-generated certificates based on the course-pacing.
 """
+import logging
+
 from celery.task import task
+from django.db.models.signals import post_save
 from django.dispatch import receiver
 from opaque_keys.edx.keys import CourseKey
 
-from certificates.models import CertificateGenerationCourseSetting
+from .config import waffle
+from certificates.models import CertificateGenerationCourseSetting, CertificateWhitelist
+from certificates.tasks import generate_certificate
 from openedx.core.djangoapps.models.course_details import COURSE_PACING_CHANGE
 
 
+log = logging.getLogger(__name__)
+
+
+@receiver(post_save, sender=CertificateWhitelist, dispatch_uid="append_certificate_whitelist")
+def _listen_for_certificate_whitelist_append(sender, instance, **kwargs):  # pylint: disable=unused-argument
+    if waffle.waffle().is_enabled(waffle.SELF_PACED_ONLY) or waffle.waffle().is_enabled(waffle.INSTRUCTOR_PACED_ONLY):
```

Comment:
What if it's an instructor-paced course and only the `SELF_PACED_ONLY` switch is enabled?